src/Compilers/Core/Portable/InternalUtilities/EnumerableExtensions.cs

```diff
@@ -752,6 +752,30 @@ private static IEnumerable<IList<T>> TransposeInternal<T>(this IEnumerable<IEnum
 
             return result;
         }
+
+        internal static IEnumerable<T> Shuffle<T>(this IEnumerable<T> source, Random? random = null)
+        {
+            if (random is null)
+            {
+#if NET6_0_OR_GREATER
+                random = Random.Shared;
+#else
+                random = new Random();
+#endif
+            }
+
+            var builder = ArrayBuilder<T>.GetInstance(source.Count());
```

Comment:
It looks like the only caller of `GetAnalyzersForTypeNames` passes an `ImmutableSortedSet` always. Can we move that signature to not use `IEnumerable`, so we can avoid the call to `Count`? I would also prefer this to not be an extension method: it doesn't seem like a general-purpose extension method that should be in the completion list for any `IEnumerable` in the roslyn project.

Reply:
I couldn't find a good location for a shared helper method like this, so made it a local function